src/almanack/processing/generate_data.py

```diff
@@ -0,0 +1,151 @@
+"""
+This module generates data for GitHub Repositories
+"""
+
+import pathlib
+import shutil
+import tempfile
+from datetime import datetime, timezone
+from typing import Optional, Tuple
+
+import pygit2
+
+from .calculate_entropy import calculate_aggregate_entropy, calculate_normalized_entropy
+from .git_operations import clone_repository, get_commits, get_edited_files
+
+
+def generate_whole_repo_data(repo_path: str) -> None:
```

Comment:
Consider the word choice here in the context of addressing the following questions. This might also be useful information which you could expand upon in the docstring.
- When would we generate part repo data (instead of whole)?
- When data is mentioned, what type of data?